pkg/retry/retry.go

```diff
@@ -0,0 +1,35 @@
+package retry
+
+import (
+	"math"
+	"time"
+
+	"github.com/grafana/pdc-agent/pkg/random"
+)
+
+type Opts struct {
+	MaxBackoff     time.Duration
+	InitialBackoff time.Duration
+}
+
+func Forever(opts Opts, f func() error) {
```

Comment:
Can we add a comment to this function please? Especially covering the edge cases: I think it looks like, if InitialBackoff is 0, then it doesnt increase exponentially (did I get that right?)

Reply:
Yeah, that's right. Added a comment.